net/net.go

```diff
@@ -16,7 +16,128 @@ limitations under the License.
 
 package net
 
-import "net"
+import (
+	"fmt"
+	"net"
+)
+
+// ParseCIDRs parses a list of cidrs and return error if any is in valid.
+// order is maintained
+func ParseCIDRs(cidrsString []string) ([]*net.IPNet, error) {
+	cidrs := make([]*net.IPNet, 0, len(cidrsString))
+	for _, cidrString := range cidrsString {
+		_, cidr, err := net.ParseCIDR(cidrString)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse cidr value:%q with error:%v", cidrString, err)
+		}
+		cidrs = append(cidrs, cidr)
+	}
+	return cidrs, nil
+}
+
+// IsDualStackIPs returns if a slice of ips is:
+// - all are valid ips
+// - at least one ip from each family (v4 or v6)
+// - an error is returned if length is less than 2
+// - an error is returned if duplicate ips are found
```

Comment:
This is low cohesion - duplicateness has nothing to do with dual-stack-ness.

I'm already not super happy with this API as a general-purpose interface, can we make the duplicate-checking a separate function?

Reply:
what I am trying to get at is that it feels like there are primitive operations that are plausibly general-purpose and can be composed by caller code into whatever combinations they need.  I'm not going to belabor the overall point because it doesn't matter THAT much, but I envisioned something like:

```
var ips []string // gets initialized somehow.
byFamily := utilnet.IPStringsByFamily(ip)
for k, v := range byFamily {
  dd, err := utilnet.DedupIPs(v)
  if err != nil { ... }
  if len(dd) != len(v) { ... }
}
```

Reply:
I am good with adding a `dedup` general purpose function. But i would argue that `duplicate` cidr or IPs is an error case. checking for duplicates makes this `unit` the smallest possible unit. calling style can be:
```
ips := getIPsSomeHow()
deDupped := dedupIPs() // currently not provided by the util packege
// at this stage ips are valid
if IsDualStackIPs(..){

}
```

as a caller, i know that `IsDualStackIPs` will return true `if and only if` it is dualstack, there is no `true even if it has duplicate` or `true even if has nils`

Reply:
I would say either:

a) assert that we transparently handle duplicates by silently de-duping (in which case IsDualStack doesn't care about dups), or...

b) assert that suplicate-ness is an error, make each caller check for dups (in which case IsDualStack doesn't care about dups)

duplicateness is orthogonal to dual-stackness, let's not jam them together just because they tend to both be checked by callers.

Reply:
I want with a). duplicates are silently ignored.

Reply:
comment still says dusp and >= 2